core/src/builtin_widgets/global_anchor.rs

```diff
@@ -0,0 +1,214 @@
+use crate::{prelude::*, ticker::FrameMsg};
+use std::rc::Rc;
+
+#[derive(Declare, Query)]
+pub struct GlobalAnchor {
+  #[declare(builtin, default)]
+  global_anchor: Anchor,
+}
+
+impl ComposeChild for GlobalAnchor {
+  type Child = Widget;
+  #[inline]
+  fn compose_child(this: impl StateWriter<Value = Self>, child: Self::Child) -> impl WidgetBuilder {
+    fn eq(f1: Anchor, f2: Anchor) -> bool { f1.x == f2.x && f1.y == f2.y }
+    fn_widget! {
+      let wnd = ctx!().window();
+      let tick_of_layout_ready = wnd
+        .frame_tick_stream()
+        .filter(|msg| matches!(msg, FrameMsg::LayoutReady(_)));
+
+      let mut child = @$child {};
+      let wid = child.lazy_id();
+      watch!(($this.get_global_anchor(), $child.layout_size()))
+        .sample(tick_of_layout_ready)
+        .subscribe(move |_| {
+          let size = $child.layout_size();
+          let wnd_size = wnd.size();
+          let base = wnd.map_to_global(Point::zero(), wid.assert_id());
+          let Anchor {x, y} = $this.get_global_anchor();
+          let anchor = Anchor {
+            x: x.map(|x| match x {
+              HAnchor::Left(x) => x - base.x,
+              HAnchor::Right(x) => (wnd_size.width - x) - size.width - base.x,
+            }).map(HAnchor::Left),
+            y: y.map(|y| match y {
+              VAnchor::Top(y) => y - base.y,
+              VAnchor::Bottom(y) => (wnd_size.height - y) - size.height - base.y,
+            }).map(VAnchor::Top),
+          };
+          if !eq($child.anchor, anchor) {
+            $child.write().anchor = anchor;
+          }
+        });
+      child
+        .widget_build(ctx!())
+        .attach_state_data(this, ctx!())
+    }
+  }
+}
+
+impl GlobalAnchor {
+  fn get_global_anchor(&self) -> Anchor { self.global_anchor }
+}
+
+fn bind_h_anchor(
+  this: &impl StateWriter<Value = GlobalAnchor>,
+  wnd: &Rc<Window>,
+  relative: HAnchor,
+  base: f32,
+) {
+  let size = wnd.size();
+  let anchor = match relative {
+    HAnchor::Left(x) => HAnchor::Left(base + x),
+    HAnchor::Right(x) => HAnchor::Right((size.width - base) + x),
+  };
+  if this.read().global_anchor.x != Some(anchor) {
+    this.write().global_anchor.x = Some(anchor);
+  }
+}
+
+fn bind_v_anchor(
+  this: &impl StateWriter<Value = GlobalAnchor>,
+  wnd: &Rc<Window>,
+  relative: VAnchor,
+  base: f32,
+) {
+  let size = wnd.size();
+  let anchor = match relative {
+    VAnchor::Top(x) => VAnchor::Top(base + x),
+    VAnchor::Bottom(x) => VAnchor::Bottom((size.height - base) + x),
+  };
+  if this.read().global_anchor.y != Some(anchor) {
+    this.write().global_anchor.y = Some(anchor);
+  }
+}
+
+impl<W> FatObj<W> {
+  /// Anchor the widget's horizontal position to the left edge of the widget
+  /// with WidgetId wid .
+  pub fn relative_to_left(
+    &mut self,
+    relative: HAnchor,
+    wid: &LazyWidgetId,
+    ctx: &BuildCtx,
+  ) -> impl Subscription {
+    let this = self.get_builtin_global_anchor(ctx).clone_writer();
```

Comment:
we can remove the `ctx` parameter requirement, because construct the global anchor not really depend on the context.

Reply:
not only for get_builtin_global_anchor, but also need to get the window.